pkg/nfd-master/nfd-master.go

```diff
@@ -417,6 +421,85 @@ func (m *nfdMaster) SetLabels(c context.Context, r *pb.SetLabelsRequest) (*pb.Se
 			klog.Errorf("failed to advertise labels: %v", err)
 			return &pb.SetLabelsReply{}, err
 		}
+		if !m.args.NoTaint {
+
+			// ensure we delete the old leftover taints
+			labelReply, err := m.deleteTaints(c, r)
+			if err != nil {
+				return labelReply, err
+			}
```

Comment:
Hey, just popped into my mind: maybe we should remove the old nfd-generated taints irrespective of the `--no-taint` setting. E.g. if one tries this (taints) out but then decides to switch the tainting off, this would ensure that old stale taints will be cleaned up.

WDYT?

Reply:
Ow, very good catch @marquiz. I totally missed that scenario. Yes, I definitely agree that we should take care of cleaning the taints  in that situation too. Thanks a lot. I will fix it.